integration/cluster_api_endpoints_test.go

```diff
@@ -0,0 +1,223 @@
+// +build integration
+
+package integration_test
+
+import (
+	"encoding/json"
+	"errors"
+	"fmt"
+	"io/ioutil"
+	"os"
+
+	awseks "github.com/aws/aws-sdk-go/service/eks"
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/ginkgo/extensions/table"
+	. "github.com/onsi/gomega"
+
+	. "github.com/weaveworks/eksctl/integration/matchers"
+	. "github.com/weaveworks/eksctl/integration/runner"
+
+	"github.com/weaveworks/eksctl/pkg/ctl/cmdutils"
+
+	api "github.com/weaveworks/eksctl/pkg/apis/eksctl.io/v1alpha5"
+)
+
+const (
+	createCluster    = `Create`
+	updateCluster    = `Update`
+	deleteCluster    = `Delete`
+	endpointPubTmpl  = `EndpointPublicAccess: %v`
+	endpointPrivTmpl = `EndpointPrivateAccess: %v`
+)
+
+func setEndpointConfig(cfg *api.ClusterConfig, privateAccess, publicAccess bool) {
+	cfg.VPC.ClusterEndpoints.PrivateAccess = &privateAccess
+	cfg.VPC.ClusterEndpoints.PublicAccess = &publicAccess
+}
+
+func generateName(prefix string) string {
+	if clusterName == "" {
+		clusterName = cmdutils.ClusterName("", "")
+	}
+	return fmt.Sprintf("%v-%v", prefix, clusterName)
+}
+
+func setMetadata(cfg *api.ClusterConfig, name, region string) {
+	cfg.Metadata.Name = name
+	cfg.Metadata.Region = region
+}
+
+func printSeparator(op, clname string) {
+	fmt.Println("\n==================================================================")
+	fmt.Printf("%v cluster %v\n", op, clname)
+	fmt.Println("==================================================================")
+}
+
+var _ = Describe("(Integration) Create and Update Cluster with Endpoint Configs", func() {
+
+	type EndpointAccessCase struct {
+		Name    string
+		Private bool
+		Public  bool
+		Type    string
+		Output  string
```

Comment:
`Output`, `Delete`, and `fun` seem unused

Reply:
Yes.  Experiments that I abandoned.

Reply:
Done.